ospd_openvas/daemon.py

```diff
@@ -1401,20 +1401,40 @@ def exec_scan(self, scan_id: str):
                     )
 
                     if scan_db.host_is_finished(openvas_scan_id):
-                        self.report_openvas_scan_status(
-                            scan_db, scan_id, current_host
+                        # Check that the host status is updated. As it has
+                        # finished, its stauts should be DEAD or FINISHED.
+                        # Otherwise it is considered an error. It tries 3
+                        # times until considered it an error
+                        retry = 3
+                        host_progress = self.get_scan_host_progress(
+                            scan_id, current_host
                         )
-
+                        while (
+                            host_progress > -1 and host_progress < 100
+                        ) and retry > 0:
+                            self.report_openvas_scan_status(
+                                scan_db, scan_id, current_host
+                            )
+                            time.sleep(1)
+                            retry -= 1
+                            host_progress = self.get_scan_host_progress(
+                                scan_id, current_host
+                            )
```

Comment:
This will only partially solve this problem. 
1. Why was 3 (seconds) chosen? This will probably only reduce the number of scans marked as INTERRUPTED, won't it?
2. If retry reaches 0, the while loop will end, but there is no separate logic to handle this case. An error should be raised or at least the error should be logged, if the following code works fine for 'invalid'  host progress data

Reply:
1. Yes, the idea is to reduce the amount of "INTERRUPTED", based in the suspicion of a race condition in this part of the code. The host finished but the progress still not updated. It retries 3 times one second, which is different to wait 3 seconds. I think 1 second for each retries is enough time to let redis to be updated and get the last plugins launch rate 100/100. If it is a race condition, on the first retry it should be leave the while loop.
2. You are right. Solved with https://github.com/greenbone/ospd-openvas/pull/390/commits/11e833808a29c9f93a7111744a85408ae1b56960

Thanks for the review!